### PR TITLE
Use `regex-pcre`

### PR DESCRIPTION
**Pull request by guibou**
This new approach uses `regex-pcre` instead of `megaparsec`.

The main improvements are:

- a dramatic performance improvement. CPU time is reduced by 10 and memory is reduced by 2
- less dependencies.

### CODE REVIEW

File: krank.cabal [hunk 1]
```diff
@@ -46,7 +46,8 @@ test-suite krank-test
                        , hspec >= 2.7 && < 2.8
                        , PyF >= 0.8.1.0 && < 0.9
                        , krank
-                       , megaparsec
+                       , pcre-heavy
+                       , bytestring
```
Comment (gbataille) [inline krank.cabal:50]: nit: do we want to keep alphabetical ordering for easier management?

**Review by gbataille — APPROVED** [review]

File: krank.cabal (hunk 1 above)
Comment (guibou) [reply krank.cabal:50]: I had never really cared about sorted items in list as long as there is no automatic tool to ensure the policy,

I'll fix that. Thank you

File: src/Krank/Checkers/IssueTracker.hs [hunk 2]
```diff
@@ -69,33 +62,36 @@ serverDomain :: GitServer
...
-  single '/'
-  repoOwner <- takeWhile1P Nothing ('/'/=)
-  single '/'
-  repoName <- takeWhile1P Nothing ('/'/=)
-  "/issues/"
-  issueNum <- decimal
-  return $ GitIssue gitServer (pack repoOwner) (pack repoName) issueNum
+-- | This regex represents a github/gitlab issue URL
+gitRepoRe :: Regex
+gitRepoRe = [re|(?:https?://)?(?:www.)?(github.com|gitlab.com)/([^/]+)/([^/]+)/issues/([0-9]+)|]
```
Comment (guibou) [inline src/Krank/Checkers/IssueTracker.hs]: `www.` should be `www\.`

File: krank.cabal (hunk 1 above)
Comment (guibou) [reply krank.cabal:50]: Not fixed for now. I'd like to have something automatic to check for these, else I don't think there is value.

File: src/Krank/Checkers/IssueTracker.hs (hunk 2 above)
Comment (guibou) [reply src/Krank/Checkers/IssueTracker.hs]: Fixed.

Comment (guibou) [reply src/Krank/Checkers/IssueTracker.hs]: `github\.com|gitlab\.com` too.

**Pull request merged**